### ISSUE TITLE
Division by zero -> INF values -> NaN values

### ISSUE THREAD

**Issue by alexkoltun** [pytorch/vision#7607]
There is a possibility that maxc == 0 and minc != maxc, in that case "s" will end up containing INF as "cr" will be divided by zero, found it the hard way after getting NaN when using ColorJitter transform.
https://github.com/pytorch/vision/blob/d2f7486ccaef461913cdb51990ff353addf6f064/torchvision/transforms/_functional_tensor.py#L283

**Comment by pmeier** [comment]
> There is a possibility that maxc == 0 and minc != maxc

That would mean, your input image has an incorrect value range in the first place. TorchVision convention for value range is `[0, 1 if dtype.is_floating else torch.iinfo(dtype).max]`. For the two most prevalent dtypes this means:
- `torch.uint8`: `[0, 255]`
- `torch.float32`: `[0.0, 1.0]`

Since `maxc` and `minc` are the plain maximum and minimum values

https://github.com/pytorch/vision/blob/d2f7486ccaef461913cdb51990ff353addf6f064/torchvision/transforms/_functional_tensor.py#L267-L268

`maxc == 0` and `minc != maxc` must mean that `minc < 0`. 

Could you post more details about the input where you have encountered this?

**Comment by alexkoltun** [comment]
I agree that your assumption should be correct, nevertheless I was able to reproduce it using randomly generated input that is normalized between 0.0 and 1.0, it seems there are some cases that defy logic :)

Following is the snippet:
```python
from torchvision import transforms
import torch

normalize = transforms.Normalize((0.0,), (1.0,))
t = transforms.ColorJitter(brightness=0.3, contrast=0.3, saturation=0.3, hue=0.3)

rng_state = torch.load('bad_rng_state.pt')
torch.set_rng_state(rng_state)
img = normalize(torch.randn(3, 224, 224))
if torch.sum(torch.isnan(img)) != 0:
    print('img NaN')

if torch.sum(torch.isinf(img)) != 0:
    print('img INF')

if torch.sum(img < 0.0) != 0:
    print('img NEG')

error_img = img.clone()
res = t(img)
if torch.sum(torch.isnan(res)) != 0:
    print('Found error after transform')
```

Attaching the rng state as well (zipped):

[bad_rng_state.pt.zip](https://github.com/pytorch/vision/files/11524029/bad_rng_state.pt.zip)

**Comment by pmeier** [comment]
[`transforms.Normalize`](https://pytorch.org/vision/stable/generated/torchvision.transforms.Normalize.html) doesn't do what you think it is doing. It implements `(input - mean) / std` where `mean` and `std` are the first parameters, respectively. Meaning, `transforms.Normalize((0.0,), (1.0,))` computes `(input - 0.0) / 1.0 == input` and thus is a no-op.

Since you create your image with `torch.randn` it is normal (punt intended) that you get negative values and also values above the valid range. Do you have any reason to use normally distributed pixel values? If so, you need to manually rescale to the `[0.0, 1.0]` range. Most likely though you want to use `torch.rand`, which gives you uniformly sampled pixel values already in the correct range. That is what we are using everywhere as well.

Feel free to re-open if I misinterpreted anything and you think there is actually a bug.
